lib/avirwrapper.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Copyright (c) 2014 Victor Dubiniuk <victor.dubiniuk@gmail.com>
+ * This file is licensed under the Affero General Public License version 3 or
+ * later.
+ * See the COPYING-README file.
+ */
+
+namespace OCA\Files_Antivirus;
+
+use OC\Files\Storage\Wrapper\Wrapper;
+use OCA\Files_Antivirus\Scanner;
+use OCA\Files_Antivirus\Content;
+use Icewind\Streams\CallbackWrapper;
+
+use \OCP\Files\InvalidContentException;
+
+class AvirWrapper extends Wrapper{
+	
+	/**
+	 * Modes that are used for writing 
+	 * @var array 
+	 */
+	private $writingModes = array('r+', 'w', 'w+', 'a', 'a+', 'x', 'x+', 'c', 'c+');
+	
+	/**
+	 * Asyncronously scan data that are written to the file
+	 * @param string $path
+	 * @param string $mode
+	 * @return resource | bool
+	 */
+	public function fopen($path, $mode){
+		$stream = $this->storage->fopen($path, $mode);
+		if (is_resource($stream)
+				&& $this->file_exists($path) 
+				&& $this->is_file($path) 
+				&& $this->isWritingMode($mode)
+		) {
+			try {
+				$application = new \OCA\Files_Antivirus\AppInfo\Application();
```

Comment:
@VicDeo well adopting the ctor of AvirWrapper accordingly should not be an issue - right?